src/resolve/resolved-conf.c

```
@@ -385,6 +437,120 @@ int config_parse_dnssd_txt(const char *unit, const char *filename, unsigned line
+        if (r <= 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Invalid DnsStubListenExtra='%s, ignoring assignment", rvalue);
+                return 0;
+        }
+
+        if (!STR_IN_SET(word, "tcp", "udp"))
+                both = true;
+
+        if (streq(word, "tcp") || both) {
+                r = dns_stub_extra_new(&tcp);
+                if (r < 0)
+                        return log_oom();
+
+                tcp->mode = DNS_STUB_LISTENER_TCP;
+        }
+
+        if (streq(word, "udp") || both) {
+                r = dns_stub_extra_new(&udp);
+                if (r < 0)
+                        return log_oom();
+
+                udp->mode = DNS_STUB_LISTENER_UDP;
+        }
+
+        q = strrchr(rvalue, ':');
+        if (!q) {
+                s = strjoin(rvalue, ":53");
+                if (!s)
+                        return log_oom();
+
+                q = s;
+        } else
+                q = rvalue;
+
+        if (tcp) {
+                r = socket_address_parse(&tcp->address, q);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Failed to parse address in DnsStubListenExtra='%s', ignoring", rvalue);
+                        return 0;
+                }
+        }
+
+        if (udp) {
+                r = socket_address_parse(&udp->address, q);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Failed to parse address in DnsStubListenExtra='%s', ignoring", rvalue);
+                        return 0;
+                }
+        }
+
+        if (tcp) {
+                r = ordered_hashmap_ensure_put(&m->dns_extra_stub_listeners, &dns_stub_listener_extra_hash_ops, tcp, tcp);
+                if (r < 0) {
+                        if (r == -ENOMEM)
+                                return log_oom();
+
+                        log_syntax(unit, LOG_ERR, filename, line, r,
```

Please use `LOG_WARNING` for ignored errors.